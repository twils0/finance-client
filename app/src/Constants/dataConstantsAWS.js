// eslint-disable-next-line
export const actionTypes = {
  SET_AWS_STATUS: 'DATA_AUTH_AWS_STATUS',
  SET_AWS_USER: 'DATA_AUTH_SET_AWS_USER',
  RESET_AWS_STATE: 'DATA_AUTH_RESET_AWS_STATE',
};
