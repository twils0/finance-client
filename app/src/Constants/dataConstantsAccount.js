export const actionTypes = {
  SET_ACCOUNT_STATUS: 'DATA_ACCOUNT_SET_ACCOUNT_STATUS',
  SET_FIELD: 'DATA_ACCOUNT_SET_FIELD',
  SET_FIELDS: 'DATA_ACCOUNT_SET_FIELDS',
  RESET_ACCOUNT_STATE: 'DATA_ACCOUNT_RESET_ACCOUNT_STATE',
};

export const statusNames = {
  AWS_FIELDS: 'accountStatusAWSFields',
  STRIPE_FIELDS: 'accountStatusStripeFields',
  UPDATE_STRIPE_FIELDS_TOKEN: 'accountStatusUpdateStripeFieldsToken',
  UPDATE_STRIPE_FIELDS_REQUEST: 'accountStatusUpdateStripeFieldsRequest',
  UPDATE_DB_FIELDS: 'accountStatusUpdateDBFields',
  UPDATE_AWS_FIELDS: 'accountStatusUpdateAWSFields',
};

export const fieldNames = {
  NAME: 'accountName',
  EMAIL: 'accountEmail',
  EMAIL_ADDITIONAL: 'accountEmailAdditional',
  PHONE: 'accountPhone',
  NAME_ON_CARD: 'accountNameOnCard',
  PROMO_CODE: 'accountPromoCode',
  PROMO_CODE_VALID: 'accountPromoCodeValid',
  CUSTOMER: 'accountCustomer',
};
