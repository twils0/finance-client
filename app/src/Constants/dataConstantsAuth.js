export const actionTypes = {
  SET_AUTHENTICATED: 'DATA_AUTH_SET_AUTHENTICATED',
  SET_REDIRECT_URL: 'DATA_AUTH_SET_REDIRECT_URL',
  SET_AUTH_STATUS: 'DATA_AUTH_SET_AUTH_STATUS',
  SET_CODE_TYPE: 'DATA_AUTH_SET_CODE_TYPE',
  RESET_AUTH_STATE: 'DATA_AUTH_RESET_AUTH_STATE',
};

export const statusNames = {
  LOGIN: 'authStatusLogin',
  SIGN_OUT_DEVICES: 'authStatusSignOutDevices',
  LOGIN_MFA: 'authStatusLoginMFA',
  SIGN_UP: 'authStatusSignUp',
  VERIFY_PHONE: 'authStatusVerifyPhone',
  VERIFY_PHONE_CODE: 'authStatusVerifyPhoneCode',
  VERIFY_EMAIL: 'authStatusVerifyEmail',
  VERIFY_EMAIL_LINK: 'authStatusVerifyEmailLink',
  FORGOT_PASSWORD: 'authStatusForgotPassword',
  RESET_PASSWORD: 'authStatusResetPassword',
  CHANGE_PASSWORD: 'authStatusChangePassword',
  DELETE_ACCOUNT: 'authStatusDeleteAccount',
  LOGOUT: 'authStatusLogout',
};

export const codeTypeNames = {
  VERIFY_PHONE: 'verifyPhone',
  VERIFY_EMAIL: 'verifyEmail',
  VERIFY_EMAIL_ADDITIONAL: 'verifyEmailAdditional',
};
