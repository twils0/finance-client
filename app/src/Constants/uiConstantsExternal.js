export const actionTypes = {
  SET_CURRENT_FORM: 'EXTERNAL_SET_CURRENT_FORM',
  SET_FORM_HEIGHT: 'EXTERNAL_SET_FORM_HEIGHT',
  SET_INPUT_VALUE_ERROR: 'EXTERNAL_SET_INPUT_VALUE_ERROR',
  SET_BUTTON_TEXT: 'EXTERNAL_SET_BUTTON_TEXT',
};

export const formNames = {
  RESET_PASSWORD: 'resetPasswordForm',
  FORGOT_PASSWORD: 'forgotPasswordForm',
  LOGIN: 'loginForm',
  CARD: 'cardForm',
  INFO: 'infoForm',
  SIGN_UP: 'signUpForm',
  CODE_MFA_PHONE: 'codePhoneMFAForm',
  CODE_VERIFY_PHONE: 'codePhoneVerifyForm',
  CODE_VERIFY_EMAIL: 'codeEmailVerifyForm',
  DEVICE: 'deviceForm',
};

export const inputNames = {
  [formNames.RESET_PASSWORD]: {
    CODE: 'codeResetPassword',
    PASSWORD: 'passwordResetPassword',
    PASSWORD2: 'password2ResetPassword',
  },
  [formNames.FORGOT_PASSWORD]: {
    EMAIL: 'loginPasswordEmail',
  },
  [formNames.LOGIN]: {
    EMAIL: 'loginEmail',
    PASSWORD: 'loginPassword',
  },
  [formNames.CARD]: {
    NAME_ON_CARD: 'cardNameOnCard',
    PROMO_CODE: 'cardPromoCode',
    STRIPE: 'cardStripe',
  },
  [formNames.INFO]: {
    NAME: 'infoName',
    PHONE: 'infoPhone',
    EMAIL: 'infoEmail',
  },
  [formNames.SIGN_UP]: {
    PASSWORD: 'signUpPassword',
    PASSWORD2: 'signUpPassword2',
  },
  [formNames.CODE_MFA_PHONE]: {
    CODE: 'codeMFAPhoneCode',
  },
  [formNames.CODE_VERIFY_PHONE]: {
    CODE: 'codeVerifyPhoneCode',
  },
  [formNames.CODE_VERIFY_EMAIL]: {
    CODE: 'codeVerifyEmailCode',
  },
};

export const buttonNames = {
  FIRST: 'firstButton',
  SECOND: 'secondButton',
  FORGOT_PASSWORD: 'forgotPasswordButton',
  RESEND_PHONE: 'resendPhoneButton',
  RESEND_EMAIL: 'resendEmailButton',
};

export const buttonTexts = {
  RESET: 'Reset',
  CANCEL: 'Cancel',
  SEND_CODE: 'Send Code',
  RESEND_PHONE: 'Resend SMS',
  RESEND_EMAIL: 'Resend Email',
  LOG_IN: 'Log In',
  FORGOT_PASSWORD: 'Forgot Password',
  SIGN_UP: 'Sign Up',
  NEXT: 'Next',
  PREVIOUS: 'Previous',
  SUBMIT: 'Submit',
  YES: 'Yes',
  NO: 'No',
  CONTINUE: 'Continue',
};
