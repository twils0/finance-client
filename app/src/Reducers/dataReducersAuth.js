import { combineReducers } from 'redux';

import { requestStatusTypes, pathNames } from '../Constants/universalConstants';
import { actionTypes, statusNames, codeTypeNames } from '../Constants/dataConstantsAuth';

export const initialState = {
  authenticated: false,
  redirectURL: pathNames.WATCHLIST,
  status: {
    [statusNames.LOGIN]: {
      id: statusNames.LOGIN,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.SIGN_OUT_DEVICES]: {
      id: statusNames.SIGN_OUT_DEVICES,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.LOGIN_MFA]: {
      id: statusNames.LOGIN_MFA,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.SIGN_UP]: {
      id: statusNames.SIGN_UP,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.VERIFY_PHONE]: {
      id: statusNames.VERIFY_PHONE,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.VERIFY_PHONE_CODE]: {
      id: statusNames.VERIFY_PHONE_CODE,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.VERIFY_EMAIL]: {
      id: statusNames.VERIFY_EMAIL,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.VERIFY_EMAIL_LINK]: {
      id: statusNames.VERIFY_EMAIL_LINK,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.FORGOT_PASSWORD]: {
      id: statusNames.FORGOT_PASSWORD,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.RESET_PASSWORD]: {
      id: statusNames.RESET_PASSWORD,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.CHANGE_PASSWORD]: {
      id: statusNames.CHANGE_PASSWORD,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.LOGOUT]: {
      id: statusNames.LOGOUT,
      status: requestStatusTypes.SUCCESS,
    },
    [statusNames.DELETE_ACCOUNT]: {
      id: statusNames.DELETE_ACCOUNT,
      status: requestStatusTypes.SUCCESS,
    },
  },
  codeTypes: {
    [codeTypeNames.VERIFY_PHONE]: {
      id: codeTypeNames.VERIFY_PHONE,
      needed: true,
    },
    [codeTypeNames.VERIFY_EMAIL]: {
      id: codeTypeNames.VERIFY_EMAIL,
      needed: true,
    },
    [codeTypeNames.VERIFY_EMAIL_ADDITIONAL]: {
      id: codeTypeNames.VERIFY_EMAIL_ADDITIONAL,
      needed: true,
    },
  },
};

export const authenticated = (state = initialState.authenticated, action) => {
  switch (action.type) {
    case actionTypes.SET_AUTHENTICATED:
      return action.payload.authenticated;
    case actionTypes.RESET_AUTH_STATE:
      return initialState.authenticated;
    default:
      return state;
  }
};

export const redirectURL = (state = initialState.redirectURL, action) => {
  switch (action.type) {
    case actionTypes.SET_REDIRECT_URL:
      return action.payload.redirectURL;
    default:
      return state;
  }
};

export const status = (state = initialState.status, action) => {
  switch (action.type) {
    case actionTypes.SET_AUTH_STATUS:
      return {
        ...state,
        [action.payload.id]: {
          ...state[action.payload.id],
          ...action.payload,
        },
      };
    case actionTypes.RESET_AUTH_STATE:
      return initialState.status;
    default:
      return state;
  }
};

export const codeTypes = (state = initialState.codeTypes, action) => {
  switch (action.type) {
    case actionTypes.SET_CODE_TYPE:
      return {
        ...state,
        [action.payload.id]: {
          ...state[action.payload.id],
          ...action.payload,
        },
      };
    case actionTypes.RESET_AUTH_STATE:
      return initialState.codeTypes;
    default:
      return state;
  }
};

const dataReducersAuth = combineReducers({
  authenticated,
  redirectURL,
  status,
  codeTypes,
});

export default dataReducersAuth;
