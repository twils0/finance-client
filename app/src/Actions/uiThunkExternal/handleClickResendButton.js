import { formNames } from '../../Constants/uiConstantsExternal';

import resendButtonCodeVerifyPhone from './resendButtonCodeVerifyPhone';

const handleClickResendButton = (payload) => {
  if (!Object.prototype.hasOwnProperty.call(payload, 'history')) {
    throw new Error(`Please enter a value for the 'history' key - ${JSON.stringify(payload)}`);
  }

  return (dispatch, getState) => {
    const state = getState();
    const { forms } = state.ui.external;

    switch (forms.current) {
      case formNames.CODE_VERIFY_PHONE: {
        const { history } = payload;

        dispatch(resendButtonCodeVerifyPhone({ history }));
        break;
      }
      default:
        break;
    }

    return null;
  };
};

export default handleClickResendButton;
