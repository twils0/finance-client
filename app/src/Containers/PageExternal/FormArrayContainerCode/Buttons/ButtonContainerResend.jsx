import React from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router-dom';
import { connect } from 'react-redux';

import { formNames, buttonNames } from '../../../../Constants/uiConstantsExternal';

import handleClickResendButton from '../../../../Actions/uiThunkExternal/handleClickResendButton';

import ButtonResend from '../../../../Components/PageExternal/FormArrayCode/Buttons/ButtonResend';

class ButtonContainerResend extends React.Component {
  handleClick = (clickEvent) => {
    clickEvent.preventDefault();
    const { handleClickButton, history } = this.props;

    handleClickButton({ history });
  };

  render() {
    const {
      forms, width, buttonTextPhone, buttonTextEmail,
    } = this.props;
    let text = null;

    if (forms.current === formNames.CODE_VERIFY_PHONE) {
      text = buttonTextPhone;
    } else {
      text = buttonTextEmail;
    }

    return (
      <ButtonResend
        currentForm={forms.current}
        width={width}
        text={text}
        handleClick={this.handleClick}
      />
    );
  }
}

ButtonContainerResend.propTypes = {
  history: PropTypes.object.isRequired,
  width: PropTypes.number.isRequired,
  buttonTextPhone: PropTypes.string.isRequired,
  buttonTextEmail: PropTypes.string.isRequired,
  forms: PropTypes.object.isRequired,
  handleClickButton: PropTypes.func.isRequired,
};

const mapStateToProps = (state, ownProps) => ({
  history: ownProps.history,
  width: state.ui.external.utilities.buttons.width,
  buttonTextPhone: state.ui.external.buttons[buttonNames.RESEND_PHONE].text,
  buttonTextEmail: state.ui.external.buttons[buttonNames.RESEND_EMAIL].text,
  forms: state.ui.external.forms,
});

const mapDispatchToProps = dispatch => ({
  handleClickButton: payload => dispatch(handleClickResendButton(payload)),
});

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(ButtonContainerResend));
