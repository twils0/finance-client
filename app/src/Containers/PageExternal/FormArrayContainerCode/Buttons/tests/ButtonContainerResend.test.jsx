import React from 'react';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { shallow } from 'enzyme';

import { initialState as initialStateAuth } from '../../../../../Reducers/dataReducersAuth';
import { initialState as initialStateUIExternal } from '../../../../../Reducers/uiReducersExternal';
import { formNames, buttonNames } from '../../../../../Constants/uiConstantsExternal';

import ButtonContainerResend from '../ButtonContainerResend';

import handleClickResendButton from '../../../../../Actions/uiThunkExternal/handleClickResendButton';

const middleware = [thunk];
const mockStore = configureMockStore(middleware);

jest.mock('../../../../../Actions/uiThunkExternal/handleClickResendButton', () => jest.fn());
handleClickResendButton.mockReturnValue(() => null);

const history = { test: 'testHistory' };

const clickEvent = {
  preventDefault: jest.fn(),
};

const shallowComponent = store =>
  shallow(<ButtonContainerResend />, {
    context: {
      store,
      router: {
        history,
        route: {
          location: {},
          match: {},
        },
      },
    },
  })
    .dive()
    .dive()
    .dive();

describe('Containers', () => {
  describe('PageExternal', () => {
    describe('FormArrayContainerCode', () => {
      describe('Buttons', () => {
        describe('ButtonContainerResend', () => {
          afterEach(() => {
            handleClickResendButton.mockReset();
          });

          it('shallow renders correctly and button works, with code verify phone', async () => {
            const stateBeforeUIExternal = JSON.parse(JSON.stringify(initialStateUIExternal));
            stateBeforeUIExternal.forms.current = formNames.CODE_VERIFY_PHONE;

            const phoneText = stateBeforeUIExternal.buttons[buttonNames.RESEND_PHONE].text;

            const store = mockStore({
              data: {
                auth: initialStateAuth,
              },
              ui: {
                external: stateBeforeUIExternal,
              },
            });

            const wrapper = shallowComponent(store);

            const instance = wrapper.instance();
            const props = wrapper.props();

            instance.handleClick(clickEvent);

            expect(wrapper).toMatchSnapshot();
            expect(props.text).toEqual(phoneText);
            expect(handleClickResendButton).toBeCalledWith({
              history,
            });
          });

          it('shallow renders correctly, with code verify email', async () => {
            const stateBeforeUIExternal = JSON.parse(JSON.stringify(initialStateUIExternal));
            stateBeforeUIExternal.forms.current = formNames.CODE_VERIFY_EMAIL;

            const emailText = stateBeforeUIExternal.buttons[buttonNames.RESEND_EMAIL].text;

            const store = mockStore({
              data: {
                auth: initialStateAuth,
              },
              ui: {
                external: stateBeforeUIExternal,
              },
            });

            const wrapper = shallowComponent(store);

            const props = wrapper.props();

            expect(wrapper).toMatchSnapshot();
            expect(props.text).toEqual(emailText);
          });
        });
      });
    });
  });
});
