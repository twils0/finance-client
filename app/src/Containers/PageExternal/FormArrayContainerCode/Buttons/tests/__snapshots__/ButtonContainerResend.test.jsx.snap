// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Containers PageExternal FormArrayContainerCode Buttons ButtonContainerResend shallow renders correctly and button works, with code verify phone 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ButtonContainerResend
    buttonTextEmail="Resend Email"
    buttonTextPhone="Resend SMS"
    forms={
      Object {
        "cardForm": Object {
          "height": 360,
          "id": "cardForm",
          "inputs": Object {
            "cardNameOnCard": Object {
              "errorMessage": "",
              "id": "cardNameOnCard",
              "parentId": "cardForm",
              "value": "",
            },
            "cardPromoCode": Object {
              "errorMessage": "",
              "id": "cardPromoCode",
              "parentId": "cardForm",
              "value": "",
            },
            "cardStripe": Object {
              "errorMessage": "",
              "id": "cardStripe",
              "parentId": "cardForm",
              "value": "",
            },
          },
          "inputsList": Array [
            "cardNameOnCard",
            "cardPromoCode",
            "cardStripe",
          ],
        },
        "codeEmailVerifyForm": Object {
          "height": 183,
          "id": "codeEmailVerifyForm",
          "inputs": Object {
            "codeVerifyEmailCode": Object {
              "errorMessage": "",
              "id": "codeVerifyEmailCode",
              "parentId": "codeEmailVerifyForm",
              "value": "",
            },
          },
          "inputsList": Array [
            "codeVerifyEmailCode",
          ],
        },
        "codePhoneMFAForm": Object {
          "height": 183,
          "id": "codePhoneMFAForm",
          "inputs": Object {
            "codeMFAPhoneCode": Object {
              "errorMessage": "",
              "id": "codeMFAPhoneCode",
              "parentId": "codePhoneMFAForm",
              "value": "",
            },
          },
          "inputsList": Array [
            "codeMFAPhoneCode",
          ],
        },
        "codePhoneVerifyForm": Object {
          "height": 183,
          "id": "codePhoneVerifyForm",
          "inputs": Object {
            "codeVerifyPhoneCode": Object {
              "errorMessage": "",
              "id": "codeVerifyPhoneCode",
              "parentId": "codePhoneVerifyForm",
              "value": "",
            },
          },
          "inputsList": Array [
            "codeVerifyPhoneCode",
          ],
        },
        "current": "codePhoneVerifyForm",
        "deviceForm": Object {
          "height": 146,
          "id": "deviceForm",
          "inputs": Object {},
          "inputsList": Array [],
        },
        "forgotPasswordForm": Object {
          "height": 220,
          "id": "forgotPasswordForm",
          "inputs": Object {
            "loginPasswordEmail": Object {
              "errorMessage": "",
              "id": "loginPasswordEmail",
              "parentId": "forgotPasswordForm",
              "value": "",
            },
          },
          "inputsList": Array [],
        },
        "infoForm": Object {
          "height": 280,
          "id": "infoForm",
          "inputs": Object {
            "infoEmail": Object {
              "errorMessage": "",
              "id": "infoEmail",
              "parentId": "infoForm",
              "value": "",
            },
            "infoName": Object {
              "errorMessage": "",
              "id": "infoName",
              "parentId": "infoForm",
              "value": "",
            },
            "infoPhone": Object {
              "errorMessage": "",
              "id": "infoPhone",
              "parentId": "infoForm",
              "value": "",
            },
          },
          "inputsList": Array [
            "infoName",
            "infoPhone",
            "infoEmail",
          ],
        },
        "list": Array [
          "resetPasswordForm",
          "forgotPasswordForm",
          "loginForm",
          "cardForm",
          "infoForm",
          "signUpForm",
          "codePhoneMFAForm",
          "codePhoneVerifyForm",
          "codeEmailVerifyForm",
          "deviceForm",
        ],
        "loginForm": Object {
          "height": 222,
          "id": "loginForm",
          "inputs": Object {
            "loginEmail": Object {
              "errorMessage": "",
              "id": "loginEmail",
              "parentId": "loginForm",
              "value": "",
            },
            "loginPassword": Object {
              "errorMessage": "",
              "id": "loginPassword",
              "parentId": "loginForm",
              "value": "",
            },
          },
          "inputsList": Array [
            "loginEmail",
            "loginPassword",
          ],
        },
        "resetPasswordForm": Object {
          "height": 339,
          "id": "resetPasswordForm",
          "inputs": Object {
            "codeResetPassword": Object {
              "errorMessage": "",
              "id": "codeResetPassword",
              "parentId": "resetPasswordForm",
              "value": "",
            },
            "password2ResetPassword": Object {
              "errorMessage": "",
              "id": "password2ResetPassword",
              "parentId": "resetPasswordForm",
              "value": "",
            },
            "passwordResetPassword": Object {
              "errorMessage": "",
              "id": "passwordResetPassword",
              "parentId": "resetPasswordForm",
              "value": "",
            },
          },
          "inputsList": Array [
            "codeResetPassword",
            "passwordResetPassword",
            "password2ResetPassword",
          ],
        },
        "signUpForm": Object {
          "height": 247,
          "id": "signUpForm",
          "inputs": Object {
            "signUpPassword": Object {
              "errorMessage": "",
              "id": "signUpPassword",
              "parentId": "signUpForm",
              "value": "",
            },
            "signUpPassword2": Object {
              "errorMessage": "",
              "id": "signUpPassword2",
              "parentId": "signUpForm",
              "value": "",
            },
          },
          "inputsList": Array [
            "signUpPassword",
            "signUpPassword2",
          ],
        },
      }
    }
    handleClickButton={[Function]}
    history={
      Object {
        "test": "testHistory",
      }
    }
    location={Object {}}
    match={Object {}}
    width={300}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "currentForm": "codePhoneVerifyForm",
      "handleClick": [Function],
      "text": "Resend SMS",
      "width": 300,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "currentForm": "codePhoneVerifyForm",
        "handleClick": [Function],
        "text": "Resend SMS",
        "width": 300,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "context": Object {
      "router": Object {
        "history": Object {
          "test": "testHistory",
        },
        "route": Object {
          "location": Object {},
          "match": Object {},
        },
      },
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
    "hydrateIn": undefined,
  },
}
`;

exports[`Containers PageExternal FormArrayContainerCode Buttons ButtonContainerResend shallow renders correctly, with code verify email 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ButtonContainerResend
    buttonTextEmail="Resend Email"
    buttonTextPhone="Resend SMS"
    forms={
      Object {
        "cardForm": Object {
          "height": 360,
          "id": "cardForm",
          "inputs": Object {
            "cardNameOnCard": Object {
              "errorMessage": "",
              "id": "cardNameOnCard",
              "parentId": "cardForm",
              "value": "",
            },
            "cardPromoCode": Object {
              "errorMessage": "",
              "id": "cardPromoCode",
              "parentId": "cardForm",
              "value": "",
            },
            "cardStripe": Object {
              "errorMessage": "",
              "id": "cardStripe",
              "parentId": "cardForm",
              "value": "",
            },
          },
          "inputsList": Array [
            "cardNameOnCard",
            "cardPromoCode",
            "cardStripe",
          ],
        },
        "codeEmailVerifyForm": Object {
          "height": 183,
          "id": "codeEmailVerifyForm",
          "inputs": Object {
            "codeVerifyEmailCode": Object {
              "errorMessage": "",
              "id": "codeVerifyEmailCode",
              "parentId": "codeEmailVerifyForm",
              "value": "",
            },
          },
          "inputsList": Array [
            "codeVerifyEmailCode",
          ],
        },
        "codePhoneMFAForm": Object {
          "height": 183,
          "id": "codePhoneMFAForm",
          "inputs": Object {
            "codeMFAPhoneCode": Object {
              "errorMessage": "",
              "id": "codeMFAPhoneCode",
              "parentId": "codePhoneMFAForm",
              "value": "",
            },
          },
          "inputsList": Array [
            "codeMFAPhoneCode",
          ],
        },
        "codePhoneVerifyForm": Object {
          "height": 183,
          "id": "codePhoneVerifyForm",
          "inputs": Object {
            "codeVerifyPhoneCode": Object {
              "errorMessage": "",
              "id": "codeVerifyPhoneCode",
              "parentId": "codePhoneVerifyForm",
              "value": "",
            },
          },
          "inputsList": Array [
            "codeVerifyPhoneCode",
          ],
        },
        "current": "codeEmailVerifyForm",
        "deviceForm": Object {
          "height": 146,
          "id": "deviceForm",
          "inputs": Object {},
          "inputsList": Array [],
        },
        "forgotPasswordForm": Object {
          "height": 220,
          "id": "forgotPasswordForm",
          "inputs": Object {
            "loginPasswordEmail": Object {
              "errorMessage": "",
              "id": "loginPasswordEmail",
              "parentId": "forgotPasswordForm",
              "value": "",
            },
          },
          "inputsList": Array [],
        },
        "infoForm": Object {
          "height": 280,
          "id": "infoForm",
          "inputs": Object {
            "infoEmail": Object {
              "errorMessage": "",
              "id": "infoEmail",
              "parentId": "infoForm",
              "value": "",
            },
            "infoName": Object {
              "errorMessage": "",
              "id": "infoName",
              "parentId": "infoForm",
              "value": "",
            },
            "infoPhone": Object {
              "errorMessage": "",
              "id": "infoPhone",
              "parentId": "infoForm",
              "value": "",
            },
          },
          "inputsList": Array [
            "infoName",
            "infoPhone",
            "infoEmail",
          ],
        },
        "list": Array [
          "resetPasswordForm",
          "forgotPasswordForm",
          "loginForm",
          "cardForm",
          "infoForm",
          "signUpForm",
          "codePhoneMFAForm",
          "codePhoneVerifyForm",
          "codeEmailVerifyForm",
          "deviceForm",
        ],
        "loginForm": Object {
          "height": 222,
          "id": "loginForm",
          "inputs": Object {
            "loginEmail": Object {
              "errorMessage": "",
              "id": "loginEmail",
              "parentId": "loginForm",
              "value": "",
            },
            "loginPassword": Object {
              "errorMessage": "",
              "id": "loginPassword",
              "parentId": "loginForm",
              "value": "",
            },
          },
          "inputsList": Array [
            "loginEmail",
            "loginPassword",
          ],
        },
        "resetPasswordForm": Object {
          "height": 339,
          "id": "resetPasswordForm",
          "inputs": Object {
            "codeResetPassword": Object {
              "errorMessage": "",
              "id": "codeResetPassword",
              "parentId": "resetPasswordForm",
              "value": "",
            },
            "password2ResetPassword": Object {
              "errorMessage": "",
              "id": "password2ResetPassword",
              "parentId": "resetPasswordForm",
              "value": "",
            },
            "passwordResetPassword": Object {
              "errorMessage": "",
              "id": "passwordResetPassword",
              "parentId": "resetPasswordForm",
              "value": "",
            },
          },
          "inputsList": Array [
            "codeResetPassword",
            "passwordResetPassword",
            "password2ResetPassword",
          ],
        },
        "signUpForm": Object {
          "height": 247,
          "id": "signUpForm",
          "inputs": Object {
            "signUpPassword": Object {
              "errorMessage": "",
              "id": "signUpPassword",
              "parentId": "signUpForm",
              "value": "",
            },
            "signUpPassword2": Object {
              "errorMessage": "",
              "id": "signUpPassword2",
              "parentId": "signUpForm",
              "value": "",
            },
          },
          "inputsList": Array [
            "signUpPassword",
            "signUpPassword2",
          ],
        },
      }
    }
    handleClickButton={[Function]}
    history={
      Object {
        "test": "testHistory",
      }
    }
    location={Object {}}
    match={Object {}}
    width={300}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "currentForm": "codeEmailVerifyForm",
      "handleClick": [Function],
      "text": "Resend Email",
      "width": 300,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "currentForm": "codeEmailVerifyForm",
        "handleClick": [Function],
        "text": "Resend Email",
        "width": 300,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "context": Object {
      "router": Object {
        "history": Object {
          "test": "testHistory",
        },
        "route": Object {
          "location": Object {},
          "match": Object {},
        },
      },
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
    "hydrateIn": undefined,
  },
}
`;
